iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSSubscriptionModel.swift

```diff
@@ -217,6 +217,9 @@ class OSSubscriptionModel: OSModel {
 
         // Set test_type if subscription model is PUSH
         if type == .push {
+            #if targetEnvironment(simulator)
+            self.testType = OSUIApplicationReleaseMode.UIApplicationReleaseDev.rawValue
+            #endif
```

Comment:
`self.testType` will just be overwrite below. In order to prevent that we need to use an `#else` here or some kind of early return logic.

Reply:
That is fine if we are able to properly detect that it is an AdHoc or wildcard build on the simulator I want it to be overwritten. The issue is when the release mode is unknown

Reply:
ah, that wasn't clear. I recommend a comment or explicitly only doing then logic when it is unknown then.

Reply:
I updated it to only set if the mode is unknown